service/Microsoft.Azure.Devices/RegistryManager.cs

```diff
@@ -70,6 +70,23 @@ public void Dispose()
         /// <returns>echoes back the Device object with the generated keys and etags</returns>
         public abstract Task<Device> AddDeviceAsync(Device device, CancellationToken cancellationToken);
 
+
+        /// <summary>
+        /// Adds a Device with Twin information
+        /// </summary>
+        /// <param name="device"></param>
+        /// <param name="twin"></param>
+        /// <returns></returns>
+        public abstract Task<BulkRegistryOperationResult> AddDeviceWithTwinAsync(Device device, Twin twin);
```

Comment:
These are new APIs. We will need to review and make sure all IoT Hub Service SDKs support them.
@tameraw @yzhong94 @jspaith @pierreca @prmathur-microsoft

Reply:
Also, missing Unit/E2E tests.